In [1]:
import pandas as pd
# 두 단계 밖으로 나갔다가 data 폴더로 들어가는 경로
df = pd.read_csv('../../data/transactions_hm.csv') 
df.head()

t_dat                                        customer_id  article_id  \
0  2019-11-05  3e2b60b679e62fb49516105b975560082922011dd752ec...   698328010   
1  2019-05-22  89647ac2274f54c770aaa4b326e0eea09610c252381f37...   760597002   
2  2019-05-10  2ebe392150feb60ca89caa8eff6c08b7ef1138cd6fdc71...   488561032   
3  2019-08-26  7b3205de4ca17a339624eb5e3086698e9984eba6b47c56...   682771001   
4  2019-08-10  3b77905de8b32045f08cedb79200cdfa477e9562429a39...   742400033   

      price  sales_channel_id  
0  0.016932                 2  
1  0.033881                 2  
2  0.016932                 2  
3  0.033881                 2  
4  0.003220                 1

In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 5 columns):
 #   Column            Non-Null Count    Dtype  
---  ------            --------------    -----  
 0   t_dat             1048575 non-null  str    
 1   customer_id       1048575 non-null  str    
 2   article_id        1048575 non-null  int64  
 3   price             1048575 non-null  float64
 4   sales_channel_id  1048575 non-null  int64  
dtypes: float64(1), int64(2), str(2)
memory usage: 40.0 MB


In [3]:
df.isnull().sum()

t_dat               0
customer_id         0
article_id          0
price               0
sales_channel_id    0
dtype: int64

In [4]:
df.describe()

article_id         price  sales_channel_id
count  1.048575e+06  1.048575e+06      1.048575e+06
mean   6.746672e+08  2.777915e-02      1.695412e+00
std    1.173503e+08  1.935764e-02      4.602328e-01
min    1.087750e+08  2.372880e-04      1.000000e+00
25%    6.289790e+08  1.523729e-02      1.000000e+00
50%    7.059880e+08  2.540678e-02      2.000000e+00
75%    7.500700e+08  3.388136e-02      2.000000e+00
max    8.797400e+08  5.067797e-01      2.000000e+00

In [5]:
df["sales_channel_id"].value_counts()

sales_channel_id
2    729192
1    319383
Name: count, dtype: int64

In [7]:
# 각 판매 채널의 비율
df["sales_channel_id"].value_counts(normalize=True)*100


sales_channel_id
2    69.541235
1    30.458765
Name: proportion, dtype: float64

온라인 구매건수가 더 많다.
온라인 69.5%, 오프라인이 30.5%로 온라인 구매건 수가 오프라인 구매건 수의 2배가 넘는다.

In [8]:
import pandas as pd
# 두 단계 밖으로 나갔다가 data 폴더로 들어가는 경로
df = pd.read_csv('../../data/articles_hm.csv') 
df.head()

article_id  product_code          prod_name  product_type_no  \
0   108775015        108775          Strap top              253   
1   108775044        108775          Strap top              253   
2   108775051        108775      Strap top (1)              253   
3   110065001        110065  OP T-shirt (Idro)              306   
4   110065002        110065  OP T-shirt (Idro)              306   

  product_type_name  product_group_name  graphical_appearance_no  \
0          Vest top  Garment Upper body                  1010016   
1          Vest top  Garment Upper body                  1010016   
2          Vest top  Garment Upper body                  1010017   
3               Bra           Underwear                  1010016   
4               Bra           Underwear                  1010016   

  graphical_appearance_name  colour_group_code colour_group_name  ...  \
0                     Solid                  9             Black  ...   
1                     Solid                 10             White  ...   
2                    Stripe                 11         Off White  ...   
3                     Solid                  9             Black  ...   
4                     Solid                 10             White  ...   

   department_name index_code        index_name index_group_no  \
0     Jersey Basic          A        Ladieswear              1   
1     Jersey Basic          A        Ladieswear              1   
2     Jersey Basic          A        Ladieswear              1   
3   Clean Lingerie          B  Lingeries/Tights              1   
4   Clean Lingerie          B  Lingeries/Tights              1   

   index_group_name section_no            section_name garment_group_no  \
0        Ladieswear         16  Womens Everyday Basics             1002   
1        Ladieswear         16  Womens Everyday Basics             1002   
2        Ladieswear         16  Womens Everyday Basics             1002   
3        Ladieswear         61         Womens Lingerie             1017   
4        Ladieswear         61         Womens Lingerie             1017   

   garment_group_name                                        detail_desc  
0        Jersey Basic            Jersey top with narrow shoulder straps.  
1        Jersey Basic            Jersey top with narrow shoulder straps.  
2        Jersey Basic            Jersey top with narrow shoulder straps.  
3   Under-, Nightwear  Microfibre T-shirt bra with underwired, moulde...  
4   Under-, Nightwear  Microfibre T-shirt bra with underwired, moulde...  

[5 rows x 25 columns]